pyxdsm/XDSM.py

```diff
@@ -19,744 +23,536 @@
-            - "all" : fade all blocks
-            - "connected" : fade all components connected to faded blocks (both source and target must be faded for a conncection to be faded)
-            - "none" : do not auto-fade anything
-            For connections there are two additional options:
-            - "incoming" : Fade all connections that are incoming to faded blocks.
-            - "outgoing" : Fade all connections that are outgoing from faded blocks.
-        """
-        self.systems = []
-        self.connections = []
-        self.left_outs = {}
-        self.right_outs = {}
-        self.ins = {}
-        self.processes = []
-
-        self.use_sfmath = use_sfmath
-        if optional_latex_packages is None:
-            self.optional_packages = []
-        else:
-            if isinstance(optional_latex_packages, str):
-                self.optional_packages = [optional_latex_packages]
-            elif isinstance(optional_latex_packages, list):
-                self.optional_packages = optional_latex_packages
-            else:
-                raise ValueError("optional_latex_packages must be a string or a list of strings")
-
-        self.auto_fade = {"inputs": "none", "outputs": "none", "connections": "none", "processes": "none"}
-        fade_options = ["all", "connected", "none"]
-        if auto_fade is not None:
-            if any([key not in self.auto_fade for key in auto_fade.keys()]):
-                raise ValueError(
-                    "The supplied 'auto_fade' dictionary contains keys that are not recognized. "
-                    + "valid keys are 'inputs', 'outputs', 'connections', 'processes'."
-                )
-
-            self.auto_fade.update(auto_fade)
-        for key in self.auto_fade.keys():
-            option_is_valid = self.auto_fade[key] in fade_options or (
-                key == "connections" and self.auto_fade[key] in ["incoming", "outgoing"]
+# Type definitions - these match the TikZ styles in diagram_styles
+NodeType = Literal[
+    "Optimization",
+    "SubOptimization",
+    "MDA",
+    "DOE",
+    "ImplicitFunction",
+    "Function",
+    "Group",
+    "ImplicitGroup",
+    "Metamodel",
+]
+ConnectionStyle = Literal["DataInter", "DataIO"]
+Side = Literal["left", "right"]
+AutoFadeOption = Literal["all", "connected", "none", "incoming", "outgoing"]
+
+# Valid TikZ node styles (from diagram_styles.tikzstyles)
+VALID_NODE_STYLES = {
+    "Optimization",
+    "SubOptimization",
+    "MDA",
+    "DOE",
+    "ImplicitFunction",
+    "Function",
+    "Group",
+    "ImplicitGroup",
+    "Metamodel",
+    "DataInter",
+    "DataIO",
+}
+
+
+class SystemNode(BaseModel):
+    """System node on the diagonal of XDSM diagram."""
+
+    node_name: str = Field(..., description="Unique name for the system")
```

Comment:
Some of these attribute definitions feel duplicated across the classes. Consider using a mix in pattern (still inheriting from Pydantic) which would allow reuse of the same attribute definitions across multiple classes